packages/rich-text/src/component/index.native.js

```diff
@@ -685,7 +690,13 @@ export class RichText extends Component {
 	}
 
 	componentWillUnmount() {
-		if ( this._editor.isFocused() && this.props.shouldBlurOnUnmount ) {
+		// `isNewRichTextAdded` flag is used to distinguish whether RichText was unmounted
+		// because of adding new RichText via pressing `enter`. If `false` it indicates
+		// block is removed and RichText should be blurred.
```

Comment:
Could you please add an info about the reason why we don't want to blur rich text when adding new one?

Reply:
Done, could you please test it?

Reply:
Thanks :)